### DISCUSSION TITLE
CMS Strapi: Does not get page

### DISCUSSION THREAD

**Discussion by SalahAdDin** [vercel/next.js#14851] [Help]
I'm following the [CMS Strapi Example](https://github.com/vercel/next.js/tree/canary/examples/cms-strapi), provided by @chibicode, but adapting it to my needs, it involves to use `page` folder instead of `post`, for instance.

Each time I want to see any page by slug I get the next error and it sends me to the `404`, i'm not getting any page despite i can get it in the **strapi** _graphql_ console. But it does not send a `fetch` error or something, even more, it gets the pages and urls correctly, as far as i can see it.

## To Reproduce

1. Run dev.
2. Go to `/page/education`

## Expected behavior

It should render the expected page as in the console.

## System information

- OS: Linux
- Browser Chrome
- Version of Next.js: 9.4
- Version of Node.js: 14

## Additional context
You can get the full code [here](https://github.com/SalahAdDin/frontend/tree/develop). I just followed the project structure and files, i didn't copy-paste at all, maybe i missed something.

I also want to say it could be great if you make a very deep blog explaining the example, because alas i don't get so much here, maybe that's why i cannot understand what is going on with this bug.

**Update:** 

It does not work with paths too, what I see it is API is returning a valid promise, but the client is not getting it:
This one is in the preview mode, we are getting the page with the given slug, no problem... until you see it is rendering 404 error.
![siquetengoelslug](https://user-images.githubusercontent.com/5034215/86573898-b2373080-bf7d-11ea-99d7-3758b50c0941.png)

Here, again, we can see a resolved promise for pages.
![coñoquesiexiste](https://user-images.githubusercontent.com/5034215/86573896-b19e9a00-bf7d-11ea-997b-3f16fc6bd0d8.png)

Here, we can see the promise is coming, at least it does exist, and it is resolved, but, again, the client does not get anything.
![nuncadejadeserindefinida](https://user-images.githubusercontent.com/5034215/86573901-b2cfc700-bf7d-11ea-8b6b-8bbbb567040b.png)

Again, pages exist in the resolved promise, but the client never gets the data.
![pagesbutnotallpages](https://user-images.githubusercontent.com/5034215/86573906-b3685d80-bf7d-11ea-97bd-594b9405db96.png)

It never got defined, because the data never comes.
![nuncalleanada](https://user-images.githubusercontent.com/5034215/86573893-b06d6d00-bf7d-11ea-87be-4041b5661de3.png)

How can i solve this? Thanks.

**Comment by remidej** [comment]
Have you tried adding an `await` on [this line](https://github.com/SalahAdDin/frontend/blob/develop/lib/api.js#L42)?

Like this:

```js
const data = await fetchAPI(`your query here`)
```

**Reply by SalahAdDin** [reply]
done, and i still have the same problem: no data.

**Reply by SalahAdDin** [reply]
@remi2j actually, if you see the original, it has no `await`: https://github.com/vercel/next.js/blob/85057e5417924931199add735a07872b5fe3d455/examples/cms-strapi/lib/api.js#L43

**Reply by SalahAdDin** [reply]
I think i got the problem: i forgot to bring the `slug` field from the **API**, and even when the page was gotten, as the `slug` is ever undefined, it would ever give me a `404` error.
